api/campaigns/class-campaign-data-utils.php

```diff
@@ -27,6 +27,23 @@ public static function is_donor( $client_data ) {
 		return ! empty( $client_data['donations'] );
 	}
 
+	/**
+	 * Compare page referrer to a list of referrers.
+	 *
+	 * @param string $page_referrer_url Referrer to compare to.
+	 * @param string $referrers_list_string Comma-separated referrer domains list.
+	 */
+	public static function does_referrer_match( $page_referrer_url, $referrers_list_string ) {
+		$referer_domain = parse_url( $page_referrer_url, PHP_URL_HOST ); // phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url, WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		// Handle the 'www' prefix – assume `www.example.com` and `example.com` referrers are the same.
+		$referer_domain_alternative = strpos( $referer_domain, 'www.' ) === 0 ? substr( $referer_domain, 4 ) : "www.$referer_domain";
```

Comment:
This made me realize that any subdomain other than `www` won't be considered a match. e.g. if I set my segment to exclude `google.com`, and the referrer is `https://gmail.google.com`, it won't be considered a match and the campaign will still be displayed. Also true of the regular referrer option. Should we clarify that URLs with subdomains must be specified in the field if they want them to match the segment?

Reply:
Now I wonder if we should maybe disregard the subdomain - just check the domain. Referral from `images.example.com` is still a referral from `example.com`, right?

Reply:
Yeah, that's what I was wondering. My gut says we should just look at the root domain and ignore subdomains.

Reply:
Right, done in d16574b